api/grpcserver/post_service.go

```diff
@@ -52,11 +55,29 @@ func NewPostService(log *zap.Logger) *PostService {
 	}
 }
 
+// AllowConnections sets if the grpc service accepts new incoming connections from post services.
+func (s *PostService) AllowConnections(allow bool) {
+	s.clientMtx.Lock()
+	defer s.clientMtx.Unlock()
+	s.allowConnections = allow
+}
+
+// connectionAllowed returns if the grpc service accepts new incoming connections from post services.
+func (s *PostService) connectionAllowed() bool {
+	s.clientMtx.Lock()
+	defer s.clientMtx.Unlock()
+	return s.allowConnections
+}
+
 // Register is called by the PoST service to connect with the node.
 // It creates a bidirectional stream that is kept open until either side closes it.
 // The other functions on this service are called by services of the node to send
 // requests to the PoST node and receive responses.
 func (s *PostService) Register(stream pb.PostService_RegisterServer) error {
+	if !s.connectionAllowed() {
+		return status.Error(codes.PermissionDenied, "connection not allowed: node has no coinbase set in config")
```

Comment:
This is a different endpoint. `SmesherService.StartSmeshing` will give information on why it fails. `PostService.Register` is called by post services when they connect to a node. The goal of this change is to not allow them to connect (by returning an error) when no coinbase is set in the nodes configuration. This error here will be logged by the node and by the post service.

Without this change the post service would connect normally when no coinbase is set but during the cyclegap the ATX builder wouldn't start building ATXs (because of the missing coinbase).

At the same time not setting a coinbase in the config is valid. A non-smeshing node has no coinbase and the only difference between it and a multi-smeshing node is if a coinbase is set or not 🙂